bin/cron/pardot_import

```diff
@@ -197,19 +188,189 @@ def main
+        log "Prospect total operations so far: #{num_operations}"
+        num_operations_last_print = num_operations
+      end
       prospects = []
     end
   end
 
-  # submit any batch remainder
-  submit_create_batch prospects unless prospects.empty?
+  # Submit any batch remainder
+  submit_prospect_batch(prospects, config) unless prospects.empty?
+  num_operations += prospects.length
+  log "Contact #{config[:operation_name]} pass completed. #{num_operations} total operations."
+end
+
+# Create or update a batch of prospects
+# @param prospects [Array<Hash>] array of hashes of prospect data
+# @param config [Hash] hash of config params to use to control create vs update behavior
+# @returns [Nokogiri::XML] XML response from Pardot
+def submit_prospect_batch(prospects, config)
+  # Build the URL containing prospect data. (Prospect data is sent as a JSON blob in a
+  # query parameter)
+  url = build_batch_prospects_url(prospects, config)
+
+  # Build array of the emails for to create in Pardot
+  prospect_emails = prospects.collect{ |x| x["email"] }.to_a
+  malformed_emails = []
+
+  num_submitted = prospects.length
+
+  # Post the data to create or update a batch of prospects
+  doc = post_with_auth_retry(url)
+  status = doc.xpath('/rsp/@stat').text
+
+  # Look for errors within the batch and handle them individually.
+  if status != "ok"
+    num_removed = 0
+    doc.xpath('/rsp/errors/*').each do |node|
+      error_text = node.text
+      if error_text.eql?("Invalid prospect email address")
+        # Identifier of contact with error is zero-based index
+        prospect_identifier = node.attr("identifier").to_i
+        # Adjust contact index for any we have already removed in this batch
+        prospect_identifier -= num_removed
+
+        malformed_email = prospect_emails[prospect_identifier]
+        log "Pardot reported email \"#{malformed_email}\" as malformed, marking as malformed in our DB"
+        # Add the rejected email to our list of malformed emails
+        malformed_emails << malformed_email
+        prospect_emails.delete_at(prospect_identifier)
+        num_removed += 1
+      else
+        error_text = "Unknown error" if error_text.empty?
+        log doc.to_s
+        log error_text
+        raise "Error in Pardot response: #{error_text}"
+      end
+    end
+  end
+
+  log "Completed Pardot prospect #{config[:operation_name]} batch call. #{num_submitted} submitted, #{prospect_emails.length} succeeded, #{malformed_emails.length} rejected as malformed"
+
+  # Mark Pardot sync time of contacts in our database
+  PEGASUS_DB[:contact_rollups].where("email in ?", prospect_emails).update("pardot_sync_at=NOW()") unless prospect_emails.empty?
+  # Mark any email addresses rejected by Pardot as malformed so we don't keep trying to fruitlessly create them forever
+  PEGASUS_DB[:contact_rollups].where("email in ?", malformed_emails).update("email_malformed=true") unless malformed_emails.empty?
```

Comment:
Can the update use hash syntax? E.g., I think we prefer `update(email_malformed: true)`. Also directly above.